lib/util/SortableSet.js

```diff
@@ -94,4 +92,16 @@ module.exports = class SortableSet extends Set {
 		this._cacheOrderIndependent.set(fn, newData);
 		return newData;
 	}
-};
+
+	_invalidateCache(all) {
+		if(this._cache !== null)
+			this._cache.clear();
+		if(all && this._cacheOrderIndependent !== null)
```

Comment:
Unroll the `_invalidateCache` method into `_invalidateCache` and `_invalidateOrderedCache` to get rid of the `all` argument for better performance.

yes `this._cache.clear();` will be duplicated in both functions.